src/pint/residuals.py

```diff
@@ -123,6 +125,26 @@ def resids_value(self):
         else:
             return self.resids.to_value(self.unit)
 
+    @property
+    def dof(self):
+        """Return number of degrees of freedom for the model."""
+        if self._is_combined:
+            raise RuntimeError(
```

Comment:
AttributeError would be cleaner here, though it might cause weirdness with `@property`.